### PR TITLE
port sx127x driver to new phy.h

### PR DESCRIPTION
**Pull request by glennergeerts**

### CODE REVIEW

**Comment by pnunes30** [comment]
Thanks for this PR!
Some time ago, when I rebased the new_phy branch, I realized idd that since the phy layer was tied to the DLL layer, it was problematic to invoke the phy APIs directly (as intended by the engineering_mode code). 
I see now that this part has been refactored to let the engineering_mode deal directly with the phy layer. That's great!
I see also that the logic regarding the different status of the packet in queue disappeared. Why not, this was not necessarily useful ;)
The only thing for which I'm cautious is the rx and tx callback being handled in the ISR context. Except for the refill callback, I would recommend to post a task in the sx127x driver to invoke such callbacks. For a gateway, it is idd important to not delay the RF processing after receiving/transmitting a packet, the application execution time is not necessarily compliant with the real time constraint of the radio processing.
More specific comments will follow in the files directly.

File: stack/modules/d7ap/phy.c [hunk 1]
```diff
@@ -473,6 +482,8 @@ error_t phy_start_rx(channel_id_t* channel, syncword_class_t syncword_class, rx_
     received_callback = rx_cb;
     // TODO error handling EINVAL, EOFF
 
+    hw_radio_is_idle();
+
```
[the remark this answers is not in the document]
Comment (LOorts-Aloxy) [reply stack/modules/d7ap/phy.c]: You're right, was hesitant about this myself. Changing this right away.

[the remark this answers is not in the document]
Comment (pnunes30) [reply stack/modules/d7ap/phy.c]: Thanks for the patch. This seems far better like this!

**Comment by glennergeerts** [comment]
merged in master

**Pull request closed without merging**